src/pages/Details.js

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import { useParams, useNavigate } from 'react-router-dom';
+import { useSelector } from 'react-redux';
+import { BiArrowBack } from 'react-icons/bi';
+import './styles/Details.css';
+
+function Details() {
+  const { name } = useParams();
+  const { data } = useSelector((state) => state.country);
+  const countryDetails = data.find((country) => country.name === name);
+  const navigation = useNavigate();
+
+  if (!countryDetails) {
+    return (
+      <div className="not-found">
+        <h2>Country Not Found</h2>
+        <button
+          type="button"
+          onClick={() => navigation('/')}
+          className="back-btn"
+        >
+          <BiArrowBack />
+        </button>
+      </div>
+    );
+  }
+
+  return (
+    <div className="details-container">
+      <div className="details-country">
+        <div className="background">
+          <button
+            type="button"
+            onClick={() => navigation('/')}
+            className="back-btn"
+          >
+            <BiArrowBack />
+          </button>
+        </div>
+        <h2 className="header">
+          {countryDetails.name}
+          &apos;s Details
+        </h2>
+      </div>
+      <div className="container">
+        <img src={countryDetails.flag} alt={countryDetails.name} />
+        <div className="flex-container">
+          <h2 className="country-name-details">{countryDetails.name}</h2>
+          <h3 className="country-details-off">
+            Official Name:
+            {' '}
+            {countryDetails.officialName}
+          </h3>
+          <h4 className="country-details">
+            Capital City:
+            {' '}
+            {countryDetails.capital}
+          </h4>
+          <h4 className="country-details">
+            Continent:
+            {' '}
+            {countryDetails.continent}
+          </h4>
+          <h4 className="country-details">
+            Population:
+            {' '}
+            {countryDetails.population}
+          </h4>
+          <h4 className="country-details">
+            Area:
+            {countryDetails.area}
+          </h4>
+          <h4 className="country-details">
+            Timezone:
+            {countryDetails.timezone[0]}
+          </h4>
+        </div>
+      </div>
```

Comment:
- [ ] Great job so far on your app. You've really put in a lot of works in there. However, you need to work on the details page. That's the only block on your PR's approval. The home page on the mobile looks very great now, but work on the details page. Take a look at how it displays on the mobile as observed by the previous reviewers. :point_down: 

### Your Homepage (Mobile)
![your-mobile-view](https://github.com/siddghosh108/React-Country-Capstone/assets/26597432/cd400251-69d5-44fa-b25b-27530b707f02)

### Your Details Page (Mobile)
![details-page](https://github.com/siddghosh108/React-Country-Capstone/assets/26597432/8356bf3f-fd15-4878-9291-4c26badba398)

### Screen record
[app-demo.webm](https://github.com/siddghosh108/React-Country-Capstone/assets/26597432/e2a072d0-ccc3-4467-8f12-b9c6fb9b725d)


- [OPTIONAL] The arrow link suggested by the previous reviewer supposed to be the link to the country's detail page. I would recommend you to make it behave as such.